CRM/Case/BAO/Case.php

```diff
@@ -556,12 +556,13 @@ public static function getCases($allCases = TRUE, $params = [], $context = 'dash
       return $getCount ? 0 : $casesList;
     }
 
+    $type = CRM_Utils_Array::value('type', $params, 'upcoming');
```

Comment:
You do know that @colemanw is on the rampage against this function? `$params['type'] ?? 'upcoming'` should save you from his ire....

Reply:
I think this is mergeable with this change.

A unit test would be a nice-to-have - it might only require setting up a single case with no future activities and then calling getCases twice.

Reply:
@eileenmcnaughton @demeritcowboy 

Should we merge this and add test case at later stage?

Reply:
Can you make the Utils_Array change noted above? And for test here's something to get you started:

```diff
--- a/tests/phpunit/CRM/Case/BAO/CaseTest.php
+++ b/tests/phpunit/CRM/Case/BAO/CaseTest.php
@@ -1377,4 +1377,28 @@ class CRM_Case_BAO_CaseTest extends CiviUnitTestCase {
     }
   }

+  /**
+   * Test that if there's only recently performed activities in the system
+   * and no future ones then it still shows on dashboard.
+   */
+  public function testOnlyRecent() {
+    $loggedInUserId = $this->createLoggedInUser();
+    $clientId = $this->individualCreate([], 0, TRUE);
+    $caseObj = $this->createCase($clientId, $loggedInUserId, ['start_date' => date('Y-m-d', strtotime('-2 years'))]);
+    // quickie hack to make them all completed
+    CRM_Core_DAO::executeQuery("UPDATE civicrm_case_activity ca INNER JOIN civicrm_activity a ON a.id = ca.activity_id SET a.status_id = 2 WHERE ca.case_id = %1", [1 => [$caseObj->id, 'Integer']]);
+    // Add a recent one
+    $activity = $this->callAPISuccess('Activity', 'create', [
+      'source_contact_id' => $loggedInUserId,
+      'target_contact_id' => $clientId,
+      'activity_type_id' => 'Follow up',
+      'status_id' => 'Completed',
+      'activity_date_time' => date('Y-m-d H:i:s', strtotime('-2 days')),
+      'subject' => 'backdated',
+      'case_id' => $caseObj->id,
+    ]);
+    $this->assertEquals(0, CRM_Case_BAO_Case::getCases(TRUE, ['type' => 'upcoming'], 'dashboard', TRUE));
+    $this->assertEquals(1, CRM_Case_BAO_Case::getCases(TRUE, ['type' => 'recent'], 'dashboard', TRUE));
+  }
+
 }
```

Reply:
@kurund can you make that CRM_Utils_Array change above and add in the above test?

Reply:
@kurund what do you want to do with this one? Should I put up a "reviewer's cut" alternate PR and close this PR?

Reply:
@demeritcowboy if you feel able to do a reviewers cut without too much more effort I would encourage you to - happy to review that

Reply:
Ok thanks I've added https://github.com/civicrm/civicrm-core/pull/25954